### PR TITLE
Enable IntelliSense prediction by default

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Enable intellisense prediction by default:
- when VT is not supported, `PredictionSource` is set to `None` by default
- when VT is supported and it's running with PowerShell 7.2+, `PredictionSource` is set to `HistoryAndPlugin` by default
- when VT is supported and it's running with PowerShell prior to 7.2, `PredictionSource` is set to `History` by default

## PR Checklist

- [x] PR has a meaningful title
    - Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [ ] Make sure you've added one or more new tests
- [x] Make sure you've tested these changes in terminals that PowerShell is commonly used in (i.e. conhost.exe, Windows Terminal, Visual Studio Code Integrated Terminal, etc.)
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] Documentation needed at [PowerShell-Docs](https://github.com/MicrosoftDocs/PowerShell-Docs)
        - [x] Doc Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/8943

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/PowerShell/PSReadLine/pull/3351)

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@andschwa Can you please review this PR when you get a chance? Thanks!

File: PSReadLine/Cmdlets.cs [hunk 1]
```diff
@@ -185,7 +185,11 @@ public PSConsoleReadLineOptions(string hostName)
             HistorySearchCaseSensitive = DefaultHistorySearchCaseSensitive;
             HistorySaveStyle = DefaultHistorySaveStyle;
             AnsiEscapeTimeout = DefaultAnsiEscapeTimeout;
-            PredictionSource = DefaultPredictionSource;
+            PredictionSource = usingLegacyConsole
+                ? PredictionSource.None
+                : Environment.Version.Major < 6
```
Comment (andyleejordan) [inline PSReadLine/Cmdlets.cs:190]: Just double-checking here @daxian-dbw, your PR description says:

> when VT is supported and it's running with PowerShell 7.2+, PredictionSource is set to HistoryAndPlugin by default
when VT is supported and it's running with PowerShell prior to 7.2, PredictionSource is set to History by default

But this is checking `Version.Major < 6`

So 7 and up (theoretically including 7.0 and 7.1, not just 7.2+, also any old versions of 6 that may still be out there) are going to be `HistoryAndPlugin`. Is that what you intend?

**Review by andyleejordan — CHANGES_REQUESTED** [review]

File: PSReadLine/Cmdlets.cs (hunk 1 above)
Comment (daxian-dbw) [reply PSReadLine/Cmdlets.cs:190]: It's `Environment.Version.Major`, meaning the major version of the .NET Core. PowerShell 7.2 comes with .NET 6, so if `Environment.Version.Major < 6`, the PowerShell version is lower than 7.2, and we will use `History` as the default prediction source.

Comment (andyleejordan) [reply PSReadLine/Cmdlets.cs:190]: Got it, got it. Thanks for checking!

**Review by andyleejordan — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@andschwa Thanks for the review!

**Pull request merged**